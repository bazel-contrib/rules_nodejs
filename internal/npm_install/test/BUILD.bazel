load("@build_bazel_rules_nodejs//:defs.bzl", "jasmine_node_test", "nodejs_binary")

filegroup(
    name = "node_modules",
    srcs = glob(["node_modules/**"]),
)

filegroup(
    name = "goldens",
    srcs = glob(["golden/**"]),
)

jasmine_node_test(
    name = "test",
    srcs = glob(["*.spec.js"]),
    data = [
        ":check.js",
        ":goldens",
        ":node_modules",
        ":yarn.lock",
    ],
    node_modules = ":node_modules",
    deps = ["//internal/npm_install:generate_build_file"],
)

nodejs_binary(
    name = "test.accept",
    data = [
        ":check.js",
        ":goldens",
        ":node_modules",
        ":update_golden.js",
        ":yarn.lock",
        "//internal/npm_install:generate_build_file",
    ],
    entry_point = "build_bazel_rules_nodejs/internal/npm_install/test/update_golden.js",
    install_source_map_support = False,
    node_modules = ":node_modules",
)
