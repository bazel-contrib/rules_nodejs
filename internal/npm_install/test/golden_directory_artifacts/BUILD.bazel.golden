
package(default_visibility = ["//visibility:public"])
load("@build_bazel_rules_nodejs//:index.bzl", "js_library")
exports_files([
    "node_modules/@angular/core",
    "node_modules/@gregmagolan/test-a",
    "node_modules/@gregmagolan/test-b",
    "node_modules/ajv",
    "node_modules/balanced-match",
    "node_modules/brace-expansion",
    "node_modules/co",
    "node_modules/concat-map",
    "node_modules/diff",
    "node_modules/fast-deep-equal",
    "node_modules/fast-json-stable-stringify",
    "node_modules/fs.realpath",
    "node_modules/glob",
    "node_modules/inflight",
    "node_modules/inherits",
    "node_modules/jasmine",
    "node_modules/jasmine-core",
    "node_modules/json-schema-traverse",
    "node_modules/minimatch",
    "node_modules/once",
    "node_modules/path-is-absolute",
    "node_modules/rxjs",
    "node_modules/tslib",
    "node_modules/unidiff",
    "node_modules/wrappy",
    "node_modules/zone.js",
])
js_library(
    name = "node_modules",
    package_name = "$node_modules_dir$",
    package_path = "",
    srcs = [
        "//@angular/core:core__files",
        "//@gregmagolan/test-a:test-a__files",
        "//@gregmagolan/test-b:test-b__files",
        "//ajv:ajv__files",
        "//balanced-match:balanced-match__files",
        "//brace-expansion:brace-expansion__files",
        "//co:co__files",
        "//concat-map:concat-map__files",
        "//diff:diff__files",
        "//fast-deep-equal:fast-deep-equal__files",
        "//fast-json-stable-stringify:fast-json-stable-stringify__files",
        "//fs.realpath:fs.realpath__files",
        "//glob:glob__files",
        "//inflight:inflight__files",
        "//inherits:inherits__files",
        "//jasmine:jasmine__files",
        "//jasmine-core:jasmine-core__files",
        "//json-schema-traverse:json-schema-traverse__files",
        "//minimatch:minimatch__files",
        "//once:once__files",
        "//path-is-absolute:path-is-absolute__files",
        "//rxjs:rxjs__files",
        "//tslib:tslib__files",
        "//unidiff:unidiff__files",
        "//wrappy:wrappy__files",
        "//zone.js:zone.js__files",
    ],
    deps = [
        "//@angular/core:core__contents",
        "//@gregmagolan/test-a:test-a__contents",
        "//@gregmagolan/test-b:test-b__contents",
        "//ajv:ajv__contents",
        "//balanced-match:balanced-match__contents",
        "//brace-expansion:brace-expansion__contents",
        "//co:co__contents",
        "//concat-map:concat-map__contents",
        "//diff:diff__contents",
        "//fast-deep-equal:fast-deep-equal__contents",
        "//fast-json-stable-stringify:fast-json-stable-stringify__contents",
        "//fs.realpath:fs.realpath__contents",
        "//glob:glob__contents",
        "//inflight:inflight__contents",
        "//inherits:inherits__contents",
        "//jasmine:jasmine__contents",
        "//jasmine-core:jasmine-core__contents",
        "//json-schema-traverse:json-schema-traverse__contents",
        "//minimatch:minimatch__contents",
        "//once:once__contents",
        "//path-is-absolute:path-is-absolute__contents",
        "//rxjs:rxjs__contents",
        "//tslib:tslib__contents",
        "//unidiff:unidiff__contents",
        "//wrappy:wrappy__contents",
        "//zone.js:zone.js__contents",
    ],
)
filegroup(
  name = "golden_files",
  srcs = [
    "//:BUILD.bazel",
    "//:manual_build_file_contents",
    "//:WORKSPACE",
    "//@angular/core:BUILD.bazel",
    "//@gregmagolan:BUILD.bazel",
    "//@gregmagolan/test-a/bin:BUILD.bazel",
    "//@gregmagolan/test-a:BUILD.bazel",
    "//@gregmagolan/test-a:index.bzl",
    "//@gregmagolan/test-b:BUILD.bazel",
    "//ajv:BUILD.bazel",
    "//jasmine/bin:BUILD.bazel",
    "//jasmine:BUILD.bazel",
    "//jasmine:index.bzl",
    "//rxjs:BUILD.bazel",
    "//unidiff:BUILD.bazel",
    "//zone.js:BUILD.bazel",
    "//some-target-a:BUILD.bazel",
    "//some-target-a2:BUILD.bazel",
    "//@some-scope/some-target-b:BUILD.bazel",
    "//@some-scope/some-target-b2:BUILD.bazel",
  ],
)